src/components/ProductDetail.tsx

```diff
@@ -0,0 +1,109 @@
+import { useContext } from "react";
+import { useParams } from 'react-router-dom';
+import { useNavigate } from 'react-router-dom';
+import { useFetch } from '../hooks/useFetch';
+import Box from './Box';
+import Text from './Text';
+import Image from './Image';
+import Button from './Button';
+import { ThemeContext } from "../ThemeContext";
+import Swal from 'sweetalert2';
+import 'sweetalert2/dist/sweetalert2.css';
+
+function ProductDetails() {
+    const navigate = useNavigate();
+
+    const back = () => {
+        navigate('../products');
+    };
+
+    const { id } = useParams();
+    const { data: product, loading, error, executePut } = useFetch(`products/${id}`);
+
+    if (loading) {
+        return <div>Loading...</div>;
+    }
+
+    if (error) {
+        return <div>Error: {error.message}</div>;
+    }
+
+    const { theme } = useContext( ThemeContext );
+
+
+    const setTrue = async (event) => {
```

Comment:
Since there's a lot of common code here, it could probably be it's own component. I'd also try renaming to something more specific. `setTrue` and `setFalse` are very generic and don't describe what the functions do.